converter.go

```diff
@@ -58,6 +58,13 @@ const (
 	UseOriginalTimestamp
 )
 
+type retainOriginalCol bool
```

Comment:
What do you think about alternative solution - instead of returning information about retaining/reusing column, maybe we can instead always return column (whatever it is - transformed or original) and call retain in `arrowStringRecordToColumn` if needed? The code should be simpler and responsibility is on transforming function and not caller (easier to omit!). What do you think?